module/zfs/abd.c

```diff
@@ -215,15 +221,37 @@ abd_fini(void)
 		kstat_delete(abd_ksp);
 		abd_ksp = NULL;
 	}
+}
+
+#else
 
-	kmem_cache_destroy(abd_chunk_cache);
-	abd_chunk_cache = NULL;
+struct page;
+#define	kpm_enable			1
+#define	abd_alloc_chunk() \
+	((struct page *)kmem_alloc(PAGESIZE, KM_SLEEP))
+#define	abd_free_chunk(chunk) 		kmem_free(chunk, PAGESIZE)
```

Comment:
@behlendorf Good point. How reliant is userspace to the actual running system's `PAGE_SIZE`? Can we get it in configure script, and define `PAGE_SIZE` then?

I'm now more worried about kernel's `PAGESIZE` value using the build system's kernel [value](https://github.com/zfsonlinux/spl/blob/master/include/sys/sysmacros.h#L111). `PAGE_SHIFT` is configurable on some arch, e.g. arm64 and ia64.